### PR TITLE
Docs(web-react): Add Grid demo #DS-897, Feat(web-react): Introduce GridSpan component #DS-431

### PR DESCRIPTION
**Pull request by pavelklibani**
### Introducing GridSpan Component #DS-431
We are introducing a new component called `<GridSpan />`.
It was already created in **web-twig** previously.

[Demo: Grid má ve všech technologiích stejná dema](https://jira.lmc.cz/browse/DS-897)

### Adding Grid Demo #DS-897
The demo in **web-react** is now the same as it is in **web** and **web-twig**. 
In order to achieve the same demo, I had to create a new component called `<GridSpan />` in **web-react**.

[React: Grid Span](https://jira.lmc.cz/browse/DS-431)

### CODE REVIEW

File: packages/web-react/src/components/Grid/__tests__/GridSpan.text.tsx
Comment (crishpeen) [inline packages/web-react/src/components/Grid/__tests__/GridSpan.text.tsx:1]: test.tsx?

File: packages/web-react/src/components/Grid/README.md [hunk 1]
```diff
@@ -26,4 +26,44 @@ Use Grid to build multiple column layouts. This Grid works on twelve column syst
...
+  <GridSpan over="10">10</GridSpan>
+</Grid>
+```
+
+Advanced example usage:
+
+```jsx
+<Grid elementType="ul">
+  <GridSpan elementType="li" over="10" tablet="8" desktop="6">
+    {' '}
```
Comment (crishpeen) [inline packages/web-react/src/components/Grid/README.md:54]: why the spaces here?

File: packages/web-react/src/types/grid.ts [hunk 2]
```diff
@@ -12,16 +13,41 @@ export interface GridElementTypeProps<T extends ElementType = 'div'> {
   elementType?: T | JSXElementConstructor<unknown>;
 }
 
+export interface GridSpanElementTypeProps<T extends ElementType = 'div'> {
+  /**
+   * The HTML element or React element used to render the Grid, e.g. 'div'.
```
Comment (crishpeen) [inline packages/web-react/src/types/grid.ts:18]:
```suggestion
   * The HTML element or React element used to render the GridSpan, e.g. 'div'.
```

File: packages/web-react/src/components/Grid/useGridSpanStyleProps.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const getGridSpanClasses = (
+  className: string,
+  props: SpiritGridSpanProps<ElementType>,
+  breakpoints: string[] = gridBreakpoints,
+) => {
+  const gridSpanClasses: string[] = [];
+
+  breakpoints.forEach((breakpoint) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Partial<GridProps>'.
```
Comment (crishpeen) [inline packages/web-react/src/components/Grid/useGridSpanStyleProps.ts:27]: is it actually `<GridProps>`?

File: packages/web-react/src/components/Grid/demo/GridResponsiveSpan.tsx
Comment (crishpeen) [inline packages/web-react/src/components/Grid/demo/GridResponsiveSpan.tsx:1]: isn't this gridSpanResponsive?

File: packages/web-react/src/components/Grid/README.md (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/README.md:54]: spaces removed

File: packages/web-react/src/types/grid.ts (hunk 2 above)
Comment (pavelklibani) [reply packages/web-react/src/types/grid.ts:18]: comment fixed

File: packages/web-react/src/components/Grid/useGridSpanStyleProps.ts (hunk 3 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/useGridSpanStyleProps.ts:27]: it is! fixed

File: packages/web-react/src/components/Grid/demo/GridResponsiveSpan.tsx
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/demo/GridResponsiveSpan.tsx:1]: Yes, all renamed to gridSpanResponsive

File: packages/web-react/src/components/Grid/README.md (hunk 1 above)
Comment (literat) [reply packages/web-react/src/components/Grid/README.md:54]: Really? I still see `{'  '}`. Was the comment about this?

File: packages/web-react/src/components/Grid/demo/GridSpan.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Grid from '../Grid';
+import GridSpan from '../GridSpan';
+
+const GridSpanDemo = () => (
+  <Grid>
+    <GridSpan over={2}>
+      <div className="docs-Box docs-Box--small">2</div>
```
Comment (literat) [inline packages/web-react/src/components/Grid/demo/GridSpan.tsx:8]: Please, create the `DocsBox` component the same way the `DocsSection` and use it. This component is also prepared in web-twig (https://github.com/lmc-eu/spirit-design-system/pull/1036/files#diff-c8ce68aa31e6a644232cb5f6a6763ce6b324336fd94e116373e4714638106c5a) so it makes sense to be also here.

File: packages/web-react/src/components/Grid/demo/GridItemFactory.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+type GridItemFactoryProps = {
+  items: number;
+  label: string;
+  classes?: string;
+};
+
+const GridItemFactory = ({ items, label, classes }: GridItemFactoryProps) => {
```
Comment (literat) [inline packages/web-react/src/components/Grid/demo/GridItemFactory.tsx:9]: I am not sure if the `GridItemFactory` fits here when we use it only for one demo. Maybe we can refactor it to be more general to accommodate all Grid demos.

```jsx
const gridItems = [
  { over: 4, label: 'test' },
  // ...
];

<GridItemFactory items={gridItems} />
```

Internally it should use the `DocsBox` component so we do not need to take care of the className.

File: packages/web-react/src/components/Grid/useGridSpanStyleProps.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+const gridBreakpoints = Object.keys(SpiritBreakpoints).map((breakpoint) => SpiritBreakpoints[breakpoint as unknown]);
+
+const getGridSpanClasses = (
+  className: string,
+  props: SpiritGridSpanProps<ElementType>,
+  breakpoints: string[] = gridBreakpoints,
+) => {
+  const gridSpanClasses: string[] = [];
+
+  breakpoints.forEach((breakpoint) => {
```
Comment (literat) [inline packages/web-react/src/components/Grid/useGridSpanStyleProps.ts:25]: This block of code is the same as in the `useGridStyleProps` hook. The only difference is the `--over` and `--span` modificator. Can we make a utility function for this in a separate file and only use the call to this function, please?

**Review by literat — CHANGES_REQUESTED** [review]
I have made a few suggestions to make this better. Otherwise, the examples are looking good. Good Job 👍

File: packages/web-react/src/components/Grid/README.md (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/README.md]: now it should be finally gone

File: packages/web-react/src/components/Grid/demo/GridSpan.tsx (hunk 4 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/demo/GridSpan.tsx]: Rebased and switched for `<DocsBox />`

File: packages/web-react/src/components/Grid/useGridSpanStyleProps.ts (hunk 6 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Grid/useGridSpanStyleProps.ts]: created hook `useGridClasses` and called it in functions you mentioned.

File: packages/web-react/src/components/Grid/README.md [hunk 7]
```diff
@@ -26,4 +26,43 @@ Use Grid to build multiple column layouts. This Grid works on twelve column syst
 | `UNSAFE_className` | `string`                                  | —       | ✕        | Wrapper custom class name           |
 | `UNSAFE_style`     | `CSSProperties`                           | —       | ✕        | Wrapper custom style                |
 
-For detailed information see [Grid](https://github.com/lmc-eu/spirit-design-system/blob/main/packages/web/src/scss/components/Grid/README.md) component
+For detailed information see [Grid] component
+
+## GridSpan
+
+Grid Span allows to center content over multiple grid columns.
```
Comment (literat) [inline packages/web-react/src/components/Grid/README.md:33]:
```suggestion
GridSpan allows to center content over multiple grid columns.
```
I think we should write this together, shouldn't we?

**Review by literat — APPROVED** [review]
LGTM 👍

File: packages/web-react/src/components/Grid/README.md (hunk 7 above)
Comment (literat) [reply packages/web-react/src/components/Grid/README.md:33]: We will unify this in the next PR.

**Pull request merged**